parser/src/main/java/com/facebook/coresql/sqllogictest/SqlLogicTest.java

```diff
@@ -0,0 +1,293 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.coresql.sqllogictest;
+
+import com.facebook.coresql.parser.AstNode;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.text.DecimalFormat;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.facebook.coresql.parser.ParserHelper.parseStatement;
+import static com.facebook.coresql.parser.Unparser.unparse;
+import static java.nio.file.FileVisitResult.CONTINUE;
+
+public final class SqlLogicTest
+{
+    private static AcceptedRdms currRdms;
+    private static Set<AcceptedRdms> skipIfSet;
+    private static AcceptedRdms onlyIf;
+    private static BufferedWriter writer;
+    private static FileVisitor<Path> fileVisitor;
+    private static Path startingDirectory;
+    private static int maxFilesToVisit;
+    private static int numFilesVisited;
+    private static int numStatementsParsed;
+    private static int numStatementsFound;
+
+    private enum SqlLogicTestError
+    {
+        PARSING_ERROR,
+        UNPARSING_ERROR,
+        UNPARSED_DOES_NOT_MATCH_ERROR
+    }
+
+    private enum AcceptedRdms
+    {
+        ALL,
+        ORACLE,
+        MYSQL,
+        MSSQL,
+        SQLITE,
+        POSTGRESQL
+    }
+
+    private SqlLogicTest() {}
+
+    public static void run()
+            throws IOException
+    {
+        setup("all");
+        Files.walkFileTree(startingDirectory, fileVisitor);
+        shutdown();
+    }
+
+    public static void run(String rdms)
+            throws IOException
+    {
+        setup(rdms);
+        Files.walkFileTree(startingDirectory, fileVisitor);
+        shutdown();
+    }
+
+    private static void setup(String rdms)
+            throws IOException
+    {
+        verifyAndSetRdms(rdms);
+        createFileForResults();
+        writer = new BufferedWriter(new FileWriter("src/main/java/com/facebook/coresql/sqllogictest/sqllogictest_results.txt"));
+        startingDirectory = Paths.get("src/main/java/com/facebook/coresql/sqllogictest/test_statements");
+        fileVisitor = new SqlLogicTestFileVisitor();
+        maxFilesToVisit = 10; // Note: visiting all files takes ~20 minutes on a Macbook Pro, i9, 32GB RAM
```

Comment:
Remove this.

Reply:
Removed and re-factored this as a parameter in the config file.

Done.